kratos/processes/find_intersected_geometrical_objects_process.h

```diff
@@ -7,355 +7,665 @@
 //  License:         BSD License
 //                     Kratos default license: kratos/license.txt
 //
-//  Main authors:    Pooyan dadvand
+//  Main authors:    Vicente Mataix Ferrandiz
 //
 
 #if !defined(KRATOS_FIND_INTERSECTED_GEOMETRICAL_OBJECTS_PROCESS_H_INCLUDED )
 #define  KRATOS_FIND_INTERSECTED_GEOMETRICAL_OBJECTS_PROCESS_H_INCLUDED
 
-
-
 // System includes
 #include <string>
 #include <iostream>
 
-
 // External includes
 
-
 // Project includes
 #include "processes/process.h"
-#include "includes/model_part.h"
+#include "containers/model.h"
+#include "includes/kratos_parameters.h"
 #include "spatial_containers/octree_binary.h"
 
-
 namespace Kratos
 {
-    namespace Internals {
-        class DistanceSpatialContainersConfigure
+namespace Internals {
+    /**
+     * @class DistanceSpatialContainersConfigure
+     * @ingroup KratosCore
+     * @brief This class contains the tools related with the distance spatial container cell data
+     * @details The CellNodeData is defined as an internal class
+     * @tparam TEntity The type of geometrical entity considered (if conditions or elements)
+     * @author Pooyan Dadvand
+     */
+    template<class TEntity = Element>
```

Comment:
Why is this template argument now needed? It is spread over all the derived classes that use this process.

Reply:
Because right now it only works with elements, I did this change already in #4263

Reply:
I know, this always requires to define the skin as an element based mesh. The point is that I don't know if this was done intentionally.

Reply:
In any case if you check, the python exposition remains the same. It only affects from the cpp side